### PR TITLE
Bazel: add basic build structure

### PR DESCRIPTION
**Pull request by mrtnzlml**
The purpose of this PR is to figure out how to use Bazel correctly in our monorepo and what should be the migration strategy. It is being continuously offloaded to master.

Inspiration:

- https://github.com/tbergq/bazel-monorepo
- https://github.com/jin/awesome-bazel
- https://oasisdigital.com/class/bazel
- https://github.com/GerritCodeReview/gerrit/blob/master/WORKSPACE
- https://github.com/vercel/next.js/pull/22267

### TODOs

- [x] setup Buildifier (https://github.com/bazelbuild/buildtools/blob/master/buildifier/README.md)
- [x] consider Shipit changes (do not ship `*.bazel` files) - https://github.com/adeira/universe/pull/1083
- [x] add Bazel files to `.npmignore` - https://github.com/adeira/universe/pull/1084
- [ ] auto-generate `BUILD.bazel` files (?)
- [ ] update docs (readme)
- [ ] fix CI cache issue
- [ ] run all scripts via Bazel and remove our `@babel/register` hack
- [ ] run Next.js via Bazel (https://github.com/vercel/next.js/search?q=bazel&type=issues)
- [x] use the latest Yarn and Node versions
- [ ] test multiple Node versions (?)
- [ ] get rid of `package.json` files (we need to replace the publishing process)
- [ ] verify testing of related repositories (change in `js` should test everything related?)
- [ ] do something with our `@babel/register` hack

### Possible migration strategy

- [ ] run all the tests via Bazel
- [ ] build and NPM publish all packages via Bazel (https://bazelbuild.github.io/rules_nodejs/Built-ins.html#pkg_npm)
- [ ] lint all packages via Bazel (?)
- [ ] _tktk_

### CODE REVIEW

File: .jest-bazel.config.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+// @flow strict
+
+global.__DEV__ = true;
+
+// TODO: should we rather have one per each project to isolate them properly?
```
Comment (tbergq) [inline .jest-bazel.config.js:5]: We can use jest-presets maybe?

Comment (mrtnzlml) [reply .jest-bazel.config.js:5]: How do you mean it? I was thinking about removing this global config and basically copy-pasting it everywhere to have truly independent repositories. This file creates a global dependency for all the repositories, which seems to be a bad practice in the monorepos world.

Comment (tbergq) [reply .jest-bazel.config.js:5]: I mean, you can have a jest preset at the root, and extend it or overwrite in each workspace. 

Surely, there will be some commons, right?

https://jestjs.io/docs/en/configuration#preset-string

Comment (mrtnzlml) [reply .jest-bazel.config.js:5]: Ah, I understand now. Sure, can be. :)

**Comment by mrtnzlml** [comment]
Closing in favor of: https://github.com/adeira/universe/pull/2794

**Pull request closed without merging**